src/System.Text.Json/tests/JsonReaderTests.cs

```diff
@@ -1,5 +1,6 @@
 ﻿using System.Collections.Generic;
```

Comment:
I think we need more tests, especially for invalid JSON, buffers too short, etc. Again, feel free to punt some or all of this and just file a TODO issues. I don't want to take your time away from school :-) and I want this commit to go in.